src/book/02-system/01-foundations/01-overview.mdx

```diff
@@ -22,3 +22,42 @@ After motions is hardware IO. The hardware will output the commands and input th
 All these modules are connected using NUClear which is a message passing software architecture. More can be found on NUClear [here](https://nuclear.readthedocs.io/en/latest/).
 
 ![Flow of the NUbots codebase](./images/software-overview.svg)
+
+## Roles
+
+Roles in the main codebase are the building blocks for creating a binary to run on a robot. A role file lists all the modules to be used in that binary. If a module is not included in the role file, that module will not run. All roles can be found in the [roles folder](https://github.com/NUbots/NUbots/tree/master/roles) in the first level of the repository and have the extension `.role`. For specifics about the NUClear roles system, see the [NUClear page](/system/foundations/nuclear#nuclear-roles).
```

Comment:
They're not building blocks, their more like a recipe.

The modules are the building blocks.

Reply:
> If a module is not included in the role file, that module will not run.

Change this to more positive, like "Only modules included in the role file will be run."

Reply:
I agree with Cameron - each role represents a binary itself. They're not really building blocks.

I think it would also be worth mentioning here that roles are the things you actually run, and perhaps link to the part in nubook where we say how to run a binary

Reply:
> > If a module is not included in the role file, that module will not run.
> 
> Change this to more positive, like "Only modules included in the role file will be run."

Since it takes more than just the inclusion of a module in a role file for it to be run (e.g. appropriate messages also need to be emitted) we should say something like this

"Only modules included in the role file will have the opportunity to be run."

Reply:
I did hesitate on this line because of what @Bidski says. Although it might be good to explicitly mention that messages need to be emitted, rather than implying it.

Reply:
Both updated.